core/src/utils/test/playwright/drag-element.ts

```diff
@@ -55,12 +67,83 @@ export const dragElementByYAxis = async (
-  }
+  // Drag the element.
+  await moveElement(page, startX, startY, 0, dragByY);
 
   await page.mouse.up();
 };
+
+const validateDragByX = (startX: number, dragByX: number, viewportWidth: number) => {
+  const endX = startX + dragByX;
+  // The element is being dragged past the right of the viewport.
+  if (endX > viewportWidth) {
+    const recommendedDragByX = viewportWidth - startX - 5;
+    throw new Error(
+      `The element is being dragged past the right of the viewport. Update the dragByX value to prevent going out of bounds. A recommended value is ${recommendedDragByX}.`
+    );
+  }
+
+  // The element is being dragged past the left of the viewport.
+  if (endX < 0) {
+    const recommendedDragByX = startX - 5;
+    throw new Error(
+      `The element is being dragged past the left of the viewport. Update the dragByX value to prevent going out of bounds. A recommended value is ${recommendedDragByX}.`
+    );
+  }
+};
+
+const validateDragByY = (startY: number, dragByY: number, viewportHeight: number) => {
+  const endY = startY + dragByY;
+  // The element is being dragged past the bottom of the viewport.
+  if (endY > viewportHeight) {
+    const recommendedDragByY = viewportHeight - startY - 5;
+    throw new Error(
+      `The element is being dragged past the bottom of the viewport. Update the dragByY value to prevent going out of bounds. A recommended value is ${recommendedDragByY}.`
+    );
+  }
+
+  // The element is being dragged past the top of the viewport.
+  if (endY < 0) {
+    const recommendedDragByY = startY - 5;
+    throw new Error(
+      `The element is being dragged past the top of the viewport. Update the dragByY value to prevent going out of bounds. A recommended value is ${recommendedDragByY}.`
+    );
+  }
+};
+
+const moveElement = async (page: E2EPage, startX: number, startY: number, dragByX = 0, dragByY = 0) => {
+  const steps = 10;
+  const browser = page.context().browser()!.browserType().name();
+
+  const viewport = page.viewportSize();
+  if (viewport === null) {
+    throw new Error(
+      'Cannot get viewport size. See https://playwright.dev/docs/api/class-page#page-viewport-size for more information'
+    );
+  }
+
+  validateDragByX(startX, dragByX, viewport.width);
+  validateDragByY(startY, dragByY, viewport.height);
+
+  const endX = startX + dragByX;
+  const endY = startY + dragByY;
+
+  // Drag the element.
+  for (let i = 1; i <= steps; i++) {
+    const middleX = startX + (endX - startX) * (i / steps);
+    const middleY = startY + (endY - startY) * (i / steps);
+
+    await page.mouse.move(middleX, middleY);
+
+    // Safari needs to wait for a repaint to occur before moving the mouse again.
+    if (browser === 'webkit' && i % 2 === 0) {
+      // Repainting every 2 steps is enough to keep the drag gesture smooth.
+      // Anything past 4 steps will cause the drag gesture to be flaky.
+      await page.evaluate(() => new Promise(requestAnimationFrame));
+    }
+  }
```

Comment:
Non-blocking, but something to consider: Playwright optionally fires intermediary steps with `page.mouse.move`. Have we tried using this instead of manually generating these steps? https://playwright.dev/docs/api/class-mouse#mouse-move

Reply:
Thanks for checking!